src/Core/Resolvers/CosmosQueryBuilder.cs

```diff
@@ -153,29 +164,56 @@ protected override string Build(Predicate? predicate)
         }
 
         /// <summary>
-        /// Build JOIN statements which will be used in the query.
-        /// It makes sure that the same table is not joined multiple times by maintaining a set of table names.
+        /// Resolves the operand either as a column, another predicate,
+        /// a SqlQueryStructure or returns it directly as string
         /// </summary>
-        /// <param name="joinstructure"></param>
-        /// <returns></returns>
-        private static string Build(Stack<CosmosJoinStructure> joinstructure)
+        protected new string ResolveOperand(PredicateOperand? operand)
         {
-            StringBuilder joinBuilder = new();
-
-            HashSet<DatabaseObject> tableNames = new();
-            foreach (CosmosJoinStructure structure in joinstructure)
+            if (operand == null)
             {
-                if (tableNames.Contains(structure.DbObject))
-                {
-                    continue;
-                }
-
-                joinBuilder.Append($" JOIN {structure.TableAlias} IN {structure.DbObject.FullName}");
-                tableNames.Add(structure.DbObject);
+                throw new ArgumentNullException(nameof(operand));
             }
 
-            return joinBuilder.ToString();
+            Column? c;
+            string? s;
+            Predicate? p;
+            BaseQueryStructure? sqlQueryStructure;
+            if ((c = operand.AsColumn()) != null)
+            {
+                return Build(c);
+            }
+            else if ((s = operand.AsString()) != null)
+            {
+                return s;
+            }
+            else if ((p = operand.AsPredicate()) != null)
+            {
+                return Build(p);
+            }
+            else if ((sqlQueryStructure = operand.AsCosmosQueryStructure()) is not null
```

Comment:
When would `sqlQueryStructure` not be of type `CosmosQueryStructure` when this code executes?

Reply:
It is a safety similar to what we have in `BaseSqlQueryBuilder`